data/src/main/java/com/microsoft/azure/kusto/data/UriUtils.java

```diff
@@ -10,14 +11,13 @@ private UriUtils() {
     }
 
     public static String setPathForUri(String uri, String path, boolean ensureTrailingSlash) throws URISyntaxException {
-        if (ensureTrailingSlash && !path.endsWith("/")) {
-            path += "/";
-        }
-        if (!path.startsWith("/")) {
-            path = "/" + path;
-        }
+        path = StringUtils.prependIfMissing(path, "/");
```

Comment:
@AsafMah Sorry I'm so late with this, but why did you add this line?

The `URIBuilder` will normalize the path, including if it's missing a leading "/"